neworigins/map.cpp

```diff
@@ -1058,7 +1059,12 @@ void MapBuilder::SpecializeZones(size_t continents, int continentAreaFraction) {
 
 	Awrite("Grow oceans");
 	int oceanCores = std::max(1, (int) oceans.size() / 4);
-	std::random_shuffle(oceans.begin(), oceans.end());
+
+	// C++17 technically no longer has std::random_shuffle.  It's been replaced with std::shuffle and some boilerplate
+	std::random_device rd;
+	std::mt19937 g(rd());
+	std::shuffle(oceans.begin(), oceans.end(), g);
+
```

Comment:
The changes in this file were necessitated by the move from C++11 to C++20.  These changes are semantically equivalent to the previous code.